### ISSUE TITLE
Won't lauch on GNOME Wayland session after update to 1.82.0.23250

### ISSUE THREAD

**Issue by neildarlow** [flathub/com.vscodium.codium#224]
Hi,

I've been using VSCodium under GNOME Wayland session successfully until the 1.82.0.23250 update after which it no longer launches.

Running from a terminal produces the following output:
```
neil@vivobook:~> flatpak run com.vscodium.codium
codium-wrapper: Evaluating /usr/lib/sdk/llvm16/enable.sh
codium-wrapper: Evaluating /usr/lib/sdk/rust-stable/enable.sh
codium-wrapper: Evaluating /usr/lib/sdk/vala/enable.sh
codium-wrapper: Setting up NPM packages
codium-wrapper: Setting up Cargo packages
codium-wrapper: Setting up Python packages
codium-wrapper: Setting up Ruby packages
Warning: 'ozone-platform-hint' is not in the list of known options, but still passed to Electron/Chromium.
neil@vivobook:~>

**Comment by neildarlow** [comment]
It's not a big deal to work in an Xorg session until this one is fixed too.

I was aware of another issue reporting a similar failure but as I've not been affected until now I thought that maybe this one might affect more people and generate interest.

**Comment by neildarlow** [comment]
Not for me. The window client area is shown but not the decorations, titlebar etc.

**Comment by da2ce7** [comment]
https://github.com/flathub/com.vscodium.codium/issues/221#issuecomment-1712553170

> Add this to your settings file, should be easier:
> 
> ```json
>     "window.titleBarStyle": "custom",
> ```

Fixes the issue for me :smile:

**Comment by neildarlow** [comment]
Tried both of above. With Custom TitleBar selected it works sometimes but isn't reliable.

**Comment by RalfJung** [comment]
Version `20aad2933e1ae45bc1cfd8292da8e7f1f1b72e2b0070cef81fc12cfd62e07f66` doesn't work for me either. I have to go all the way back to `be96c3ecc66eb95ae045f7bd988f93f273d0fbc5e966ba38e166cb61f7aecf76`. This is probably related to my overrides:
```
[Context]
sockets=!wayland
```

This override is required because electron-on-wayland is not currently usable for me due to https://github.com/electron/electron/issues/29345.

**Comment by neildarlow** [comment]
It will work, as is, if the titlebar style is not set to Custom i.e. not set or set to Native.

**Comment by RalfJung** [comment]
I don't have any titlebar style set. It is still not working.

**Comment by noonsleeper** [comment]
I already reverted all the Wayland changes, in the future when the support matures a little more I will reconsider again change to Wayland as default, but now the electron autodetection is buggy and cause all these problems. But I'll create some kind of dock to enable Wayland support.

**Comment by bbhtt** [comment]
> But I'll create some kind of dock to enable Wayland support.

Instead of enabling it by default I recommend adding environment variables to enable them that can be set by flatseal or whatever by people. (command line switches cannot be persistently set). See for example https://github.com/flathub/org.signal.Signal/blob/master/signal-desktop.sh

Electron doesn't default to Wayland, and enabling it creates bugs that app developers/maintainers can do nothing about.

**Comment by noonsleeper** [comment]
Hey, this was fixed on #305, then I'll close this. If you have another problem, please don't hesitate and open a new issue.
